Csanád - chapter15 address review comments (#313)
hjwp/Book-TDD-Web-Dev-Python: chapter_15_advanced_forms.asciidoc

[...]
----

[role="less_space pagebreak-before"]
[[testing-for-stupidity]]

> .An Aside on When to Test for Developer Irrationality

Reviewer: can we go with "silliness"?   it's more the kind of thing i would say.

```suggestion
.An Aside on When to Test for Developer Silliness
```

*******************************************************************************

One of the judgement calls in testing is when you should write tests that sound

> like "check that we haven't done something really dumb".  In general, you should

Reviewer: ```suggestion
like "check that we haven't done something weird".  In general, you should
```

be wary of these.

In this case, we've written a test to check that you can't save duplicate items
to the same list.  Now, the simplest way to get that test to pass, the way in
which you'd write the fewest lines of code, would be to make it impossible to
save 'any' duplicate items.  That justifies writing another test, despite the

> fact that it would be quite silly or "wrong" thing for us to code.

Reviewer: ```suggestion
fact that it would be a "silly" or "wrong" thing for us to code.
```

But you can't be writing tests for every possible way we could have coded
something wrongfootnote:[Although you can come pretty close. We are covering
the essentials here, which alone is quite sufficient for writing nice,
production-ready code. Once you're not too overwhelmed by writing tests
manually, take a look at the
https://hypothesis.readthedocs.io/en/latest/[Hypothesis] library! It helps you
set up rules for generating input for your tests, essentially covering many
more test scenarios than would make sense typing manually.].  If you have a

> function that adds two numbers, you can write a couple of tests:

Reviewer: 👍

----

One way of putting it is that you should trust yourself not to do something

> 'deliberately' irrational, but not something 'accidentally' silly.

Reviewer: this sentence only really works if it's the same word each time, it's about the contrast between _deliberately_ and _accidentally_

```suggestion
_deliberately_ silly, but not something _accidentally_ silly.
```

[...]
----
====

> // RITA: Do you expect readers to know the term "grandparent save method"? I don't think you've used this term in the book before. Just checking.

Author: I think this should be fine, yes.

`<input>` and add a few attributes manually:

// JAN: Can't you simply add the end version of the code below? These Git views are awful to read and even worse for copy&paste ... 
// CSANAD: I actually like these. They show quite clearly what to remove and what to add and IMO the reader also learns more from
// typing these rather than just copying and pasting. If they really want to just copy and paste, then they could just open the

> // book-example repo anyway.

Reviewer: 👍

We have one unit test file for each of our key source code files.  Here's
a recap of the biggest (and highest-level) one, 'test_views' (the listing
shows just the key tests and assertions, and your order may vary):
// CSANAD: Why are we only showing ListViewTest? HomePageTest, NewListTest

> // RITA: Why is this in a box? I don't think it needs to be because it's a code block that is discussed in the preceding body text. Did you do this so it would double as the "wrap-up" box that you have at the end of most chapters? It looks a bit strange this way. 

Reviewer: i actually _did_ do it so it would be the wrap_up box.

maybe we can prefix the title with "Wrap-up: What to Test in Views" instead?

[[what-to-test-in-views]]
.What to Test in Views
******************************************************************************

> 

Reviewer: Let's have an intro sentence like:

_By way of a recap, let's see an outline of all the test methods and main assertions in our test_views.  This isn't to say you should copy-paste these exactly, more like a list of things you should at least consider testing_
